data_preparation/predictor_data_prep.r

```diff
@@ -245,205 +314,142 @@ process_indicators <- function(crop, region, start_year, end_year, crop_mask_fil
-            rast_stack[rast_stack > 100] <- NA
-            rast_stack[rast_stack < 0] <- NA
-          } else if (indicator == "ndvi") {
-            # check data_preparation/global_MOD09CMG
-            # Value range: 50 to 250 for NDVI.
-            rast_stack[rast_stack < 50] <- NA
-            rast_stack[rast_stack > 250] <- NA
-            #  To scale, apply the formula: (x - 50)/200
-            rast_stack <- (rast_stack - 50)/200
-          } else if ((indicator == "ET0") | (indicator_source == "GLDAS")) {
-            # check data_preparation/global_ETo_FAO
-            rast_stack[rast_stack == -9999] <- NA
-          } else if (grepl("Temperature", indicator, fixed=TRUE)) {
-            # temp_C = temp_K - 273.15
-            rast_stack <- rast_stack - 273.15
-          }
-
-          # replicate crop mask to match number of indicator layers
-          # we set weights for NA values to 0
-          crop_masks = rep(crop_mask, nlyr(rast_stack))
-          crop_masks[is.na(rast_stack)] = 0
-
-          # NOTE the order of multiplication matters
-          rast_stack = rast_stack * crop_masks
-
-          result = extract(rast_stack, sel_shapes, fun=sum, na.rm=TRUE, ID=FALSE)
-          result$adm_id = sel_shapes$adm_id
-
-          result_w = extract(crop_masks, sel_shapes, fun=sum, na.rm=TRUE, ID=FALSE)
-          result_w$adm_id = sel_shapes$adm_id
-
-          # Divide the result by weights. Added a tiny number to avoid division by Zero.
-          result[-ncol(result)] = result[-ncol(result)]/(result_w[-ncol(result_w)] + 1e-6)
-          result <- melt(result)
-          
-          colnames(result) <- c("adm_id", "ind_file", indicator)
-          result$date <- str_sub(result$ind_file, -8,-1)
-          result$crop_name <- crop
-          result <- result[, c("crop_name", "adm_id",
-                               "date", indicator)]
-          # print(head(result))
-          if (is.null(region_results)) {
-            region_results <- result
-          } else {
-            region_results <- rbind(region_results, result)
-          }
-        }
-        # print(head(region_results))
-      }
-      if (!dir.exists(file.path(AGML_ROOT, "R-output",
-                                crop, region, indicator))) {
-        dir.create(file.path(AGML_ROOT, "R-output",
-                             crop, region, indicator),
-                  recursive=TRUE)
+        found_raster_file <- length(file_list) > 0
       }
-      write.csv(region_results,
-                file.path(AGML_ROOT, "R-output",
-                          crop, region, indicator,
-                          paste0(indicator, "_", region, ".csv")),
-                row.names=FALSE)
+
+      # resample
+      ind_rast <- rast(file_list[[1]])
+      crop_mask <- resample(crop_mask, ind_rast, method="bilinear")
+      crop_mask <- crop(crop_mask, sel_shapes)
+      resampled_filename <- file.path(AGML_ROOT, "crop_masks", "resampled_crop_mask.tif")
```

Comment:
I think you need one per indicator, otherwise they will overwrite each other. Also, you can save time by checking if the file already exists, and if so, loading it, instead of having to resample and crop every time you run the script

Reply:
Overwriting is fine since we go through indicators sequentially.